File: app/src/main/java/edu/hebut/here/utils/ZipUtils.java
package edu.hebut.here.utils;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.nio.charset.StandardCharsets;
import java.util.zip.GZIPInputStream;
import java.util.zip.GZIPOutputStream;

public class ZipUtils {
    // 压缩字符串----"ISO-8859-1"
    private String compress(String data){
        String finalData=null;
        try{
            if (data == null || data.length() == 0) {
                return data;
            }
            //打开字节输出流
            ByteArrayOutputStream bout=new ByteArrayOutputStream();
            //打开压缩用的输出流,压缩后的结果放在bout中
            GZIPOutputStream gout=new GZIPOutputStream(bout);
            //写入待压缩的字节数组
            gout.write(data.getBytes(StandardCharsets.UTF_8));
            //完成压缩写入
            gout.finish();
            //关闭输出流
            gout.close();
            finalData=bout.toString("UTF-8");
        }catch(Exception e){
            e.printStackTrace();
        }
        return finalData;
    }
    // 解压缩
    public static String uncompress(String data) {
        String finalData=null;
        try{
            //打开字节输出流
            ByteArrayOutputStream bout = new ByteArrayOutputStream();
            ByteArrayInputStream bin = new ByteArrayInputStream(data.getBytes(StandardCharsets.UTF_8));
            //打开压缩用的输出流,压缩后的结果放在bout中
            GZIPInputStream gin=new GZIPInputStream(bin);
            //写入待压缩的字节数组
            byte[] buffer = new byte[2048];
            int n;
            while ((n = gin.read(buffer)) >= 0) {
                bout.write(buffer, 0, n);
            }
            finalData=bout.toString("UTF-8");
        }catch(Exception e){
            e.printStackTrace();
        }
        return finalData;
    }
}

File: app/src/main/java/com/lxp/here/utils/ZipUtils.kt
package com.lxp.here.utils

import java.io.ByteArrayInputStream
import java.io.ByteArrayOutputStream
import java.nio.charset.StandardCharsets
import java.util.zip.GZIPInputStream
import java.util.zip.GZIPOutputStream

/* 功能描述
 * @author lixianpeng
 * @since 2021-12-04
 */

object ZipUtils {
    fun compress(data:String) : String {
        if (data.isEmpty()) {
            return data
        }
        val byteArrayOutputStream:ByteArrayOutputStream = ByteArrayOutputStream()
        val gzipOutputStream:GZIPOutputStream = GZIPOutputStream(byteArrayOutputStream)
        gzipOutputStream.write(data.toByteArray(StandardCharsets.UTF_8))
        gzipOutputStream.finish()
        gzipOutputStream.close()
        return byteArrayOutputStream.toString("UTF-8")
    }

    fun uncompress(data: String):String {
        val byteArrayOutputStream:ByteArrayOutputStream = ByteArrayOutputStream()
        val byteArrayInputStream:ByteArrayInputStream = ByteArrayInputStream(data.toByteArray(StandardCharsets.UTF_8))
        val gzipInputStream:GZIPInputStream = GZIPInputStream(byteArrayInputStream)
        val buffer:ByteArray = ByteArray(2048)
        var n: Int
        while (gzipInputStream.read(buffer).also { n = it } >= 0) {
            byteArrayOutputStream.write(buffer, 0, n)
        }
        return byteArrayOutputStream.toString("UTF-8")
    }
}